packages/dds/tree/api-report/tree.api.md

```diff
@@ -20,7 +20,7 @@ export function adaptEnum<TScope extends string, const TEnum extends Record<stri
     readonly value: TValue;
 }) & { readonly [Property in keyof TEnum]: TreeNodeSchemaClass<ScopedSchemaName<TScope, TEnum[Property]>, NodeKind.Object, object & TreeNode & ObjectFromSchemaRecord<EmptyObject> & {
         readonly value: TEnum[Property];
-    }, object & InsertableObjectFromSchemaRecord<EmptyObject>, true, unknown> & (new () => object & TreeNode & ObjectFromSchemaRecord<EmptyObject> & {
+    }, object & {} & {}, true, unknown> & (new () => object & TreeNode & ObjectFromSchemaRecord<EmptyObject> & {
```

Comment:
Thats annoying. I wonder why that happens.

Reply:
Using our FlattenKeys util around the two mapped objects might help. Might also break things.

Reply:
I'll give it a shot

Reply:
Doing so removes 1 of the 2 empty objects, and would also require we promote `FlattenKeys` and `_InlineTrick` from `@internal` to `@public`. So probably not worth it.

Reply:
Actually, the impact on intellisense might be a reason to use this anyways (and promote the types). Interested in your thoughts.

Before:
![image](https://github.com/microsoft/FluidFramework/assets/54606601/6af8d949-0541-4771-a5ee-9e0d7c051ffb)

After:
![image](https://github.com/microsoft/FluidFramework/assets/54606601/bef3cb54-59f7-4659-b68f-44b0108f693e)

Reply:
does it break navigate to source, refactor-rename or doc comments on fields? (Those all should be controlled the same way, so I'd expect all or none to work, so no need to test each one).

Does flatten keys work for the recursive cases? In the past I've had it break them.

Assuming it works in all cases, Michael is working on getting us an "InternalTypes" scope again which would make using FlattenKeys in the public less annoying, so maybe we want to do it?

Using Flatten keys on the setup where we have both the optional and required version of fields might also work, maybe try that combo?

Reply:
Yeah, flatten keys cleans things up nicely. And navigation, intellisense docs, and refactor-rename all work correctly ✅

Reply:
It does not work for the unsafe case, though, so that was left alone.